Bug fixes and CI refactor

obalunenko opened the pull request:
1) Fix vet issue  
```
go vet ./...
# github.com/TheForgotten69/goinsta/v2
./goinsta.go:490:33: conversion from int64 to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
```

2) Fix imports
3) Fix markdown linting issues
4) Make library code reproducible 
I believe vendoring gives your application projects the most durability since the project owns every line of source code it needs to build the applications. If you want a reproducible build without needing to rely on external services (like module mirrors) and being connected to the network, vendoring is the solution.

These are other benefits of vendoring:

If dependencies are removed from the VCS or somehow proxy servers lose modules, you are covered.
Upgrading dependencies can be seen by running diffs and you maintain a history.
You will have the ability to trace and debug your dependencies and test changes if necessary.
Once you run a go mod tidy and go mod vendor your changes will be replaced.

5) Fix int overflow issue on ARM platform
```
# github.com/TheForgotten69/goinsta/v2
./media.go:1007:35: constant 8999999999 overflows int
```

6) Use https://github.com/stretchr/testify in tests to simplify code
7) Add tests for uploading media
8) Add makefile and scripts

obalunenko commented:
@TheForgotten69 could you please review my PR?

TheForgotten69 commented:
Hello @obalunenko and thank you for your contribution and this PR. It's very much appreciated.

Truth be told, I don't believe vendoring has its place in this project. I fully agree with you that it's much easier to work with if you don't have access to network or the libraries disappear but it should be the responsibility of the developer to handle these things by himself when he clones the project or works within its application.
Have these libraries backup in the cloud or in another mirror should be much easier to work with.

Could you please remove the vendoring part of your PR? I really like your other changes!

TheForgotten69 commented on .gitignore:
```diff
@@ -23,7 +23,8 @@ _testmain.go
 *.prof
 
-*.out
+# Output of the go coverage tool, specifically when used with LiteIDE
```
Didn't knew that, nice one !

TheForgotten69 commented on go.mod:
```diff
@@ -1,3 +1,11 @@
+
+require (
+	github.com/stretchr/testify v1.7.0
+	github.com/tcnksm/go-input v0.0.0-20180404061846-548a7d7a8ee8
```
Isn't it only used on examples ?

obalunenko replied:
Seems so. But that is a good practice to keep go.mod up to sync with all dependencies. In other case examples should be moved to separate module.

obalunenko commented:
Closing this PR in favor of https://github.com/TheForgotten69/goinsta/pull/8

obalunenko commented:
Hello @TheForgotten69! I appreciate your review. I removed vendor as you asked and moved the code to separate PR https://github.com/TheForgotten69/goinsta/pull/8
This I closed, the branch could be deleted
